pricegraph/src/graph/shortest_paths.rs

```diff
@@ -177,31 +139,33 @@ where
 ///
 /// The orginal source can be found here:
 /// https://docs.rs/petgraph/0.5.0/src/petgraph/algo/mod.rs.html#745-792
-fn bellman_ford<G>(
+fn bellman_ford<'a, G>(
     g: G,
     source: G::NodeId,
-) -> Result<ShortestPathGraph<G>, NegativeCycle<G::NodeId>>
+) -> Result<ShortestPathGraph<'a, G>, NegativeCycle<G::NodeId>>
 where
-    G: NodeCount + IntoNodeIdentifiers + IntoEdges + NodeIndexable,
+    G: 'a + NodeCount + IntoNodeIdentifiers + IntoEdges + NodeIndexable,
+    G::NodeId: Ord,
     G::EdgeWeight: FloatMeasure,
 {
     let mut shortest_path_graph = ShortestPathGraph::empty(g, source);
 
     // scan up to |V| - 1 times.
-    for _ in 1..g.node_count() {
+    for _ in 1..=g.node_count() - 1 {
```

Comment:
Why this change?

Reply:
Good point. It was taken from how we changed the statement in the bounded case.

Reply:
It should not overflow in the context because the function input `source` is a node of the graph.